etna/whatson/models.py

```diff
@@ -685,6 +685,33 @@ def primary_access_type(self):
         if primary_access := self.event_access_types.first():
             return primary_access.access_type
 
+    @cached_property
+    def date_time_range(self):
+        format_day_date_and_time = "%A %-d %B %Y, %H:%M"
+        format_date_only = "%-d %B %Y"
+        format_time_only = "%H:%M"
+        # One session on one date where start and end times are the same
+        # return eg. Monday 1 January 2024, 19:00
+        if (self.start_date == self.end_date) and (len(self.sessions.all()) == 1):
+            return self.start_date.strftime(format_day_date_and_time)
+        # One session on one date where there are values for both start time and end time
+        # eg. Monday 1 January 2024, 19:00–20:00 (note this uses an en dash)
+        dates_same = self.start_date.date() == self.end_date.date()
+        if (
+            dates_same
+            and (self.start_date.time() != self.end_date.time())
+            and (len(self.sessions.all()) == 1)
+        ):
+            return f"{self.start_date.strftime(format_day_date_and_time)}–{self.end_date.strftime(format_time_only)}"
+        # Multiple sessions on one date
+        # Eg. Monday 1 January 2024
+        if dates_same and len(self.sessions.all()) > 1:
+            return self.start_date.strftime("%A %-d %B %Y")
```

Comment:
Suggestion: declare variable for this format string too, for consistency.

Reply:
whoops, missed that one

Reply:
Added in https://github.com/nationalarchives/ds-wagtail/pull/1470/commits/3b4dfa3deaa36afd1c22dac630c7805595577fea